widget/contrib/gpmdp.lua

```diff
@@ -8,79 +8,96 @@
 
 local helpers             = require("lain.helpers")
 local json                = require("lain.util.dkjson")
+local watch               = require("awful.widget").watch
 local focused             = require("awful.screen").focused
-local pread               = require("awful.util").pread
 local naughty             = require("naughty")
-local wibox               = require("wibox")
 local next, getenv, table = next, os.getenv, table
 
 -- Google Play Music Desktop infos
 -- lain.widget.contrib.gpmdp
 -- requires: curl
 
 local function factory(args)
```

Comment:
Instead of `factory` you can directly go with

```lua
local gpmdp = { 
    -- [...]
}
```

See [here](https://awesomewm.org/recipes/countdown/) for an example.